src/components/Icon.tsx

```diff
@@ -0,0 +1,27 @@
+import { FunctionComponent, SyntheticEvent } from 'react'
+
+import * as icons from '@mui/icons-material'
+
+interface Icon {
+    name: keyof typeof icons
+    size: number
+    className?: string | undefined
+    color?: string
+    key?: string
+    onClick?: (event: SyntheticEvent) => void
+    variant?: 'boxed'
+}
+
+export const Icon: FunctionComponent<Icon> = ({ name, size, className = '', color, key, onClick, variant }) => {
```

Comment:
I'm seeing this console error on some pages like `/code-insights`, we should probably give key a default value. I was bringing in this [random id creator function](https://github.com/sourcegraph/about/blob/improve-syntax-highlighting/src/util/createRandomId.ts) from the Learn site in my working PR, maybe bring that into here?
<img width="1703" alt="image" src="https://user-images.githubusercontent.com/59381432/174336709-1e3b50aa-076e-4f46-9015-6181ec261731.png">

Reply:
First of all, what a helpful script. Wow!

It looks like the error is related to the presence of the `key` prop in `Icon.tsx`, because it goes away once the prop is removed from the `FunctionComponent` constant. However, we can still use `createRandomId()` to assign the random key value within the return. For instances where the `Icon` component needs a key value, such as the `map()` on `CustomCarousel` creating the Code Insights error, a key value still needs to be assigned at the function level.

See: 1273f57

Reply:
Exactly! Sounds good , thanks :)